### PR TITLE
Some improvements in GerritTriggerPluginListener

### PR DESCRIPTION
**Pull request by Jimilian**
See commit messages.

### CODE REVIEW

**Comment by Jimilian** [comment]
@sschuberth, can you take a look please?

File: src/main/java/org/jenkinsci/plugins/fluentd/FluentLoggerHolder.java [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 import org.fluentd.logger.FluentLogger;
 
+import javax.annotation.CheckForNull;
```
Comment (sschuberth) [inline src/main/java/org/jenkinsci/plugins/fluentd/FluentLoggerHolder.java]: Nit: Snouldn't this be sorted in after the line below?

File: src/main/java/org/jenkinsci/plugins/fluentd/Fluentd.java [hunk 2]
```diff
@@ -90,8 +90,15 @@ public void perform(@Nonnull Run<?, ?> build, @Nonnull FilePath workspace, @Nonn
     private boolean send(String json, String jsonFromFile, @Nonnull Run<?, ?> build, @Nonnull TaskListener listener) throws InterruptedException {
         final Map<String, String> envVars = getEnvVariables(build, listener);
 
+        FluentLogger logger = FluentLoggerHolder.getLogger();
+
+        if (logger == null) {
+            listener.error("Can't send data: fluentd logger is not initialised");
```
Comment (sschuberth) [inline src/main/java/org/jenkinsci/plugins/fluentd/Fluentd.java]: Nit: "initialised" is British English, but we should use American English "initialized" (likewise below).

File: src/main/java/org/jenkinsci/plugins/fluentd/FluentHelper.java [hunk 3]
```diff
@@ -6,14 +6,16 @@
 import org.json.simple.JSONObject;
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.util.Map;
-import java.util.logging.Logger;
+
```
Comment (sschuberth) [inline src/main/java/org/jenkinsci/plugins/fluentd/FluentHelper.java]: Nit: Superfluous addition of blank line.

**Review by sschuberth — COMMENTED** [review]
> Switch to slf4j instead of standard java logger

Why? Please add a rationale to the commit message.

**Comment by sschuberth** [comment]
> Add validation duration to the Gerrit dataset

Why? Please explain in the commit message what problem this does solve.

File: src/main/java/org/jenkinsci/plugins/fluentd/FluentLoggerHolder.java (hunk 1 above)
Comment (Jimilian) [reply src/main/java/org/jenkinsci/plugins/fluentd/FluentLoggerHolder.java]: I tried to move it in different place, but IDEA always returns it here on `Reformat code` action

File: src/main/java/org/jenkinsci/plugins/fluentd/Fluentd.java (hunk 2 above)
Comment (Jimilian) [reply src/main/java/org/jenkinsci/plugins/fluentd/Fluentd.java]: Done

File: src/main/java/org/jenkinsci/plugins/fluentd/FluentHelper.java (hunk 3 above)
Comment (Jimilian) [reply src/main/java/org/jenkinsci/plugins/fluentd/FluentHelper.java]: Done

**Review by Jimilian — COMMENTED** [review]
All requested changes are done and commit messages are updated.

File: src/main/java/org/jenkinsci/plugins/fluentd/FluentLoggerHolder.java (hunk 1 above)
Comment (sschuberth) [reply src/main/java/org/jenkinsci/plugins/fluentd/FluentLoggerHolder.java]: The you should probably reconfigure IDEA :wink: Anyway, not that important.

File: src/main/java/org/jenkinsci/plugins/fluentd/GerritTriggerPluginListener.java [hunk 4]
```diff
@@ -44,17 +51,34 @@ public void onCompleted(Result result, GerritTriggeredEvent gerritTriggeredEvent
...
+                    GERRIT_TRIGGER_COMPLETED_TAG,
+                    gerritTriggeredEvent.toString(),
+                    exception.getMessage()));
+        }
+    }
+
+    private void logEvent(String tag, GerritTriggeredEvent gerritTriggeredEvent, Map<String, Object> data) {
+        FluentLogger logger = FluentLoggerHolder.getLogger();
+        if (logger == null) {
+            LOGGER.warning("Can't send data: fluentd logger is not initialised");
```
Comment (sschuberth) [inline src/main/java/org/jenkinsci/plugins/fluentd/GerritTriggerPluginListener.java]: Hehe, you missed it 😉

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: src/main/java/org/jenkinsci/plugins/fluentd/GerritTriggerPluginListener.java (hunk 4 above)
Comment (Jimilian) [reply src/main/java/org/jenkinsci/plugins/fluentd/GerritTriggerPluginListener.java]: I'm using default one. `¯\_(ツ)_/¯`
Done.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**